components/logins/README.md

```diff
@@ -3,72 +3,134 @@
+* A **logins store** is a syncable encrypted database containing login records. In order to use the logins store,
+the application must first *unlock* it by providing a secret key (preferably obtained from an OS-level keystore
+mechanism). It can then create, read, update and delete login records from the database.
+  * If the application is connected to Firefox Sync, it can instruct the store to sync itself with the user's
+    server-side logins data. This will upload any local modifications as well as download any new logins data
+    from the server, automatically reconciling records in the case of conflict.
+
+### Examples
+- [Android integration](https://github.com/mozilla-mobile/android-components/blob/master/components/service/sync-logins/README.md)
+
+
+### API Documentation
+- TODO [Expand and update API docs](https://github.com/mozilla/application-services/issues/1747)
+
+
+## Working on the Logins component
+
+### Prerequisites
+
+To effectively work on the Logins component, you will need to be familiar with:
+
+* Our general [guidelines for contributors](../../docs/contributing.md).
+* The [core concepts](#core-concepts) for users of the component, outlined above.
+* The way we [generate ffi bindings](../../docs/howtos/building-a-rust-component.md) and expose them to
+  [Kotlin](../../docs/howtos/exposing-rust-components-to-kotlin.md) and
+  [Swift](../../docs/howtos/exposing-rust-components-to-swift.md).
+* The key ideas behind [how Firefox Sync works](../../docs/synconomicon/) and the [sync15 crate](../sync15/README.md).
+
+### Implementation Overview
+
+Logins implements encrypted storage for login records on top of SQLcipher. The storage schema is based on the one
+originally used in [Firefox for
+iOS](https://github.com/mozilla-mobile/firefox-ios/blob/faa6a2839abf4da2c54ff1b3291174b50b31ab2c/Storage/SQL/SQLiteLogins.swift),
+but with the following notable differences:
+- the queries; they've been substantially modified for our needs here.
+- how sync is performed; the version here allows syncs to complete with fewer database operations.
+- timestamps; iOS uses microseconds, where the Logins component uses milliseconds.
+
+See the header comment in [`src/schema.rs`](./src/schema.rs) for an overview of the schema.
+
+### Directory structure
 The relevant directories are as follows:
 
-- `src`: The meat of the library. This contains cross-platform rust code that
+- [`src`](./src): The meat of the library. This contains cross-platform rust code that
   implements the actual storage and sync of login records.
-- `example`: This contains example rust code for syncing, displaying, and
-  editing logins using the code in `src`.
-- `ffi`: The Rust public FFI bindings. This is a (memory-unsafe, by necessity)
-  API that is exposed to Kotlin and Swift. It leverages the `ffi_support` crate
+- [`example`](./example): This contains example rust code that implements a command-line app
```

Comment:
nit: This link is also broken. I `think` this should be `[examples](./examples)`.